Kudu.Services/Diagnostics/LogStreamManager.cs

```diff
@@ -279,21 +282,29 @@ private string ParseRequest(HttpContext context)
                 return _logPath;
             }
 
-            // in case of application or http log, we ensure directory
+            // find the first part after /logstream/ which can be application, http or a custom defined path.
             string firstPath = routePath.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries)[0];
             bool isApplication = String.Equals(firstPath, "Application", StringComparison.OrdinalIgnoreCase);
             if (isApplication)
             {
                 _enableTrace = true;
+            }
+
+            // in case of application or http log, we ensure directory
+            bool isHttp = String.Equals(firstPath, "http", StringComparison.OrdinalIgnoreCase);
+            if (isHttp || isApplication)
+            {
                 FileSystemHelpers.EnsureDirectory(Path.Combine(_logPath, firstPath));
+                return Path.Combine(_logPath, routePath);
             }
-            else
+
+            // in case a custom log file path exist
+            string path = _settings.GetValue(SettingsKeys.CustomLogStream(firstPath));
+            if (!string.IsNullOrEmpty(path))
             {
-                bool isHttp = String.Equals(firstPath, "http", StringComparison.OrdinalIgnoreCase);
-                if (isHttp)
-                {
-                    FileSystemHelpers.EnsureDirectory(Path.Combine(_logPath, firstPath));
-                }
+                var customLogPath = Path.Combine(_environment.RootPath, path.TrimStart('\\'));
```

Comment:
I only use the routepath to identify the correct settings key. It's not used for the path to the log files folder. As it could be anywhere within the site I think it makes sense that the "enitre" relative path comes from the setting.